scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -51,4 +65,51 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
             analyzer.analyze();
         };
     }
+
+    List<Issue> runExternalAnalyzers(Project project) {
+        StringBuilder newImports = new StringBuilder();
+        StringBuilder tomlDependencies = new StringBuilder();
+        int importCounter = 0;
+
+        for (ScanTomlFile.Analyzer analyzer : scanTomlFile.getAnalyzers()) {
+            String reportingSource = analyzer.getOrg() + FORWARD_SLASH + analyzer.getName();
+            String analyzerImport = IMPORT_PREFIX + reportingSource + USE_IMPORT_AS_SERVICE;
+            newImports.append(analyzerImport).append("\n");
+
+            if (analyzer.getVersion() != null) {
+                tomlDependencies.append("\n");
+                tomlDependencies.append("[[dependency]]" + "\n");
+                tomlDependencies.append("org='").append(analyzer.getOrg()).append("'\n");
+                tomlDependencies.append("name='").append(analyzer.getName()).append("'\n");
+                tomlDependencies.append("version='").append(analyzer.getVersion()).append("'\n");
+
+                if (analyzer.getRepository() != null) {
+                    tomlDependencies.append("repository='").append(analyzer.getRepository()).append("'\n");
+                }
+            }
+            importCounter++;
+        }
+
+        Module defaultModule = project.currentPackage().getDefaultModule();
+        Document mainBAL = defaultModule.document(defaultModule.documentIds().iterator().next());
```

Comment:
May be this?
```suggestion
        Document mainBal = defaultModule.document(defaultModule.documentIds().iterator().next());
```

Reply:
In Ballerina projects, the default module can have file names other than 'main.bal', so will change the variable name to a more generic term.